### DISCUSSION TITLE
Can I use the query I already fetched client-side to use it server-side?

### DISCUSSION THREAD

**Discussion by bymoe** [TanStack/query#3457] [Q&A] [answered]
So, I've went through the [SSR & Next.js](https://react-query.tanstack.com/guides/ssr) documentation page, But it does not solve my issue that:
I have a query - `blog-post` - fetched twice, first, it's fetched on the client-side. but then when I navigate back to page where I fetch the `blog-post` on the server-side using `getServerSideProps` the query is being called again on the server, even though I called it in the client-side and it's there cached!

I know it's tricky. Appreciate any help!

**Comment by TkDodo** [comment, marked as answer]
The problem is not react-query specific, it's generally per design in nextJs that `getServerSideProps` will be called on every page transition.

I think the best way around this would be to do shallow routing and additionally do a check on the server as described here: https://github.com/vercel/next.js/discussions/14855#discussioncomment-555833

**Reply by TkDodo** [reply]
but given that it's a new page, the routing doesn't actually have to be shallow. The check for `'/_next/data'` should suffice

https://nextjs.org/docs/routing/shallow-routing#caveats

**Reply by bymoe** [reply]
Thank you! that shallow routing solved it for me!
